### PR TITLE
Ericsson 6600 series

### PR DESCRIPTION
**Pull request by loopodoopo**
Support for Ericsson 6600 series

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [X] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [X] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

image

image

### CODE REVIEW

File: tests/data/ericsson-6600_6600.json [hunk 1]
```diff
@@ -4,16 +4,16 @@
             "devices": [
```
Comment (Jellyfrog) [inline tests/data/ericsson-6600_6600.json:4]: renaming ciena-saos_6500.json → tests/data/ericsson-6600_6600.json seems wrong?

Comment (loopodoopo) [reply tests/data/ericsson-6600_6600.json:4]: No, that is not correct... Restored ciena-saos_6500.json.

**Comment by Jellyfrog** [comment]
Good work here!
Only question I have is if the OS name should really be "Ericsson 6600"

**Comment by loopodoopo** [comment]
Altered OS name as per request.

**Comment by murrant** [comment]
Try not to do so many snmpgets.  One change that would help a lot is to use `$this->getCacheTable('ifName', 'IF-MIB');`  This will fetch the entire ifName column only once per discovery run.

**Comment by loopodoopo** [comment]
FYI: For this device support to work, it is important to have SNMP "Extended read access" enabled on the NE. 

image

**Comment by loopodoopo** [comment]
Thank you for the feedback @murrant. Implemented.

File: LibreNMS/OS/Ericsson6600.php [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+        $data = snmpwalk_cache_oid($this->getDeviceArray(), 'xfCarrierTermActualCapacity', [], 'XF-RADIOLINK-RLT-MIB');
+        foreach ($data as $index => $entry) {
+            $sensors[] = new WirelessSensor(
+                'rate',
+                $this->getDeviceId(),
+                '.1.3.6.1.4.1.193.81.3.4.5.1.3.1.13.' . $index,
+                'ericsson-6600',
+                $index,
+                'Rate: ' . snmp_get($this->getDeviceArray(), 'xfCarrierTermDistinguishedName' . '.' . $index, '-Oqv', 'XF-RADIOLINK-RLT-MIB'),
```
Comment (Jellyfrog) [inline LibreNMS/OS/Ericsson6600.php:74]: Could this one also be cached?

File: LibreNMS/OS/Ericsson6600.php [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+        $data = snmpwalk_cache_oid($this->getDeviceArray(), 'xfCarrierTermSNIR', [], 'XF-RADIOLINK-RLT-MIB');
+        foreach ($data as $index => $entry) {
+            $sensors[] = new WirelessSensor(
+                'snr',
+                $this->getDeviceId(),
+                '.1.3.6.1.4.1.193.81.3.4.5.1.3.1.17.' . $index,
+                'ericsson-6600',
+                $index,
+                'SNR: ' . snmp_get($this->getDeviceArray(), 'xfCarrierTermDistinguishedName' . '.' . $index, '-Oqv', 'XF-RADIOLINK-RLT-MIB'),
```
Comment (Jellyfrog) [inline LibreNMS/OS/Ericsson6600.php:52]: And this

**Comment by Jellyfrog** [comment]
Still wondering about the OS name tho. 
Is there some cli on these boxes where you could check for some os name? Maybe in some software version output

File: tests/data/ericsson-6600_6600.json [hunk 4]
```diff
@@ -0,0 +1,5711 @@
+{
+    "os": {
+        "discovery": {
+            "devices": [
+                {
+                    "sysName": "<private>",
+                    "sysObjectID": ".1.3.6.1.4.1.193.81.1.1.3",
+                    "sysDescr": "MINI-LINK 6600",
+                    "sysContact": "<private>",
+                    "version": null,
```
Comment (Jellyfrog) [inline tests/data/ericsson-6600_6600.json:10]: Could you also try look for version, hardware, serial oids?

File: LibreNMS/OS/Ericsson6600.php (hunk 2 above)
Comment (loopodoopo) [reply LibreNMS/OS/Ericsson6600.php:74]: Done

File: LibreNMS/OS/Ericsson6600.php (hunk 3 above)
Comment (loopodoopo) [reply LibreNMS/OS/Ericsson6600.php:52]: Done

File: tests/data/ericsson-6600_6600.json (hunk 4 above)
Comment (loopodoopo) [reply tests/data/ericsson-6600_6600.json:10]: I'm able to find the hardware and added it. I believe we should set "sysDescr" to "MINI-LINK 6600 series" where hardware will reflect the actual type. 

I'm unable to find the software version in the MIB. 

image

Comment (loopodoopo) [reply tests/data/ericsson-6600_6600.json:10]: image

**Comment by loopodoopo** [comment]
I can find the following in the GUI, but I'm unable to find these software releases back in SNMP: 

image

I did get the hardware in: 

image

**Comment by loopodoopo** [comment]
Any advice? I hope to get this one closed :)

File: LibreNMS/OS/Ericsson6600.php [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * @link       https://www.librenms.org
+ * @copyright  2017 Tony Murray
```
Comment (murrant) [inline LibreNMS/OS/Ericsson6600.php:21]: Hmm, I don't remember writing this code :)  Please set copywright correctly, or omit it.

**Comment by murrant** [comment]
Perhaps the OS name should be Radiolink

File: LibreNMS/OS/Ericsson6600.php (hunk 5 above)
Comment (loopodoopo) [reply LibreNMS/OS/Ericsson6600.php]: Fixed this one

**Comment by loopodoopo** [comment]
> I can find the following in the GUI, but I'm unable to find these software releases back in SNMP:
> 
> image
> 
> I did get the hardware in:
> 
> image
> 
> Looks like 6300 uses the same sysObjectID but does have different OID's for some objects and other objects missing completely:
> 
> Example, I have "xfCarrierTermActualCapacity" in 6600 and "xfTermCapacity" in 6300
> I don't have any SINR in 6300
> 
> There is also an OS in Libre called "ericsson-ml"m but this sysObjectID is completely different.
> 
> Please advise

Fixed this, 6300 series will have sys.Descr set to "MINI-LINK Traffic Node" so excluding that for the 6600 series. I believe the naming of this OS is correct.

**Review by murrant — APPROVED** [review]
Thanks @loopodoopo sorry for the delay :)

**Pull request merged**